src/_igraph/convert.c

```diff
@@ -2584,7 +2584,7 @@ int igraphmodule_PyObject_to_vector_ptr_t(PyObject* list, igraph_vector_ptr_t* v
 
   IGRAPH_VECTOR_PTR_SET_ITEM_DESTRUCTOR(vec, igraph_vector_destroy);
   while ((item = PyIter_Next(it)) != 0) {
-    subvec = igraph_Calloc(1, igraph_vector_t);
+    subvec = IGRAPH_CALLOC(1, igraph_vector_t);
```

Comment:
There is no longer any need to allocate this on the heap. Keep it on the stack.

It seems to me that there was a memory leak in the original function as the heap-allocated vector was not freed on error. https://github.com/igraph/python-igraph/issues/512  In this refactoring, this has been converted to a permanent memory leak (it happens regardless of error).